src/buffer.c

```diff
@@ -759,7 +759,8 @@ int git_buf_join(
 	ssize_t offset_a = -1;
 
 	/* not safe to have str_b point internally to the buffer */
-	assert(str_b < buf->ptr || str_b >= buf->ptr + buf->size);
+	if (buf->size)
+		assert(str_b < buf->ptr || str_b >= buf->ptr + buf->size);
```

Comment:
Curious, what exactly is the undefined behaviour here?

Reply:
From https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#ubsan-checks

> `-fsanitize=pointer-overflow`: Performing pointer arithmetic which overflows, or where either the old or new pointer value is a null pointer (or in C, when they both are).

This is because C sort of explicitly disallows that: https://stackoverflow.com/a/54233071/688337